CoffeeChat/Controller/SchedulingPlacesViewController.swift

```diff
@@ -118,25 +116,66 @@ class SchedulingPlacesViewController: UIViewController {
     override func viewDidLoad() {
         super.viewDidLoad()
         view.backgroundColor = .backgroundLightGreen
-        navigationController?.navigationBar.isHidden = true // TODO Remove this
 
+        setupViews()
+        setupForStatus()
+        setupConstraints()
+        updateNext()
+    }
+
+    // MARK: - Setup Functions
+    private func setupForStatus() {
+        if schedulingStatus == .choosing {
+            campusLocations = campusLocations.filter { matchLocations.contains($0) }
+            ctownLocations = ctownLocations.filter { matchLocations.contains($0) }
+        }
+        locationSections = [
+            .campus(campusLocations),
+            .ctown(ctownLocations)
+        ]
+
+        switch schedulingStatus {
+        case .pickingTypical:
+            titleLabel.text = "Where do you prefer?"
+        case .confirming:
+            titleLabel.text = "Confirm preferred places"
+            preselectSavedLocations()
+        case .choosing:
+            titleLabel.text = "Pick a place to meet"
+        }
+    }
+
+    private func preselectSavedLocations() {
+        selectedCampusLocations = campusLocations.filter { savedLocations.contains($0) }
+        selectedCtownLocations = ctownLocations.filter { savedLocations.contains($0) }
+
+        for campusLocation in selectedCampusLocations {
+            guard let index = campusLocations.firstIndex(of: campusLocation) else { continue }
+            locationsCollectionView.selectItem(at: IndexPath(item: index, section: 0), animated: false, scrollPosition: .top)
+        }
+        for ctownLocation in selectedCtownLocations {
+            guard let index = ctownLocations.firstIndex(of: ctownLocation) else { continue }
+            locationsCollectionView.selectItem(at: IndexPath(item: index, section: 1), animated: false, scrollPosition: .top)
+        }
+
+        updateNext()
+    }
+
+    private func setupViews() {
         titleLabel.font = ._24CircularStdMedium
-        titleLabel.text = isConfirming
-            ? "Confirm preferred places"
-            : isPicking ? "Pick a place to meet" : "Where do you prefer?"
         titleLabel.textColor = .black
         view.addSubview(titleLabel)
 
-        var scheduledTime = ""
         var amPm = ""
         if Time.isAm(time: pickedTime.time) {
```

Comment:
Can this be just condensed into `let amPm =  Time.isAm(time: pickedTime.time) ? "AM" : "PM"`. I'm not entirely sure if there are other cases where we don't want amPm to have a value..? I feel like the two should cover all the cases though.